### PR TITLE
web: Highlight line of code when start == end

### PR DESCRIPTION
**Pull request by jamestiotio**
Enable code highlighting when there is only 1 selected line of code.

### CODE REVIEW

**Comment by jamestiotio** [comment]
For additional context, I discovered these issues when experimenting with the code highlighting on Elixir and comparing it with the code highlighting on GitHub. The way the GitHub UI selects the range of lines to be highlighted is slightly different, so I only implemented minimal changes to add these features instead of refactoring too much and attempting to ensure parity with GitHub UI's behavior.

**Comment by fstachura** [comment]
Hello, thank you for the PR. As for the script misbehaving when first line > last line, IIRC this is fixed in #338, which also rewrites the logic a bit. 
As for highlighting a single line, I actually didn't add it to the original code on purpose. But maybe it is a good idea. 

> The way the GitHub UI selects the range of lines to be highlighted is slightly different, so I only implemented minimal changes to add these features instead of refactoring too much and attempting to ensure parity with GitHub UI's behavior.

Could you summarize what functionality was adopted from Github's behavior? Besides what's in the title of the PR.

**Comment by jamestiotio** [comment]
> As for the script misbehaving when first line > last line, IIRC this is fixed in #338, which also rewrites the logic a bit. As for highlighting a single line, I actually didn't add it to the original code on purpose. But maybe it is a good idea.

Sure, in that case, I can add the feature to highlight a single line on top of that PR. I can rebase my changes once that PR is merged.

> Could you summarize what functionality was adopted from Github's behavior? Besides what's in the title of the PR.

Oh, I think I should have phrased it more clearly. I meant that I **did not** add any other functionalities from GitHub UI's behavior.

GitHub UI sets both the start and the end of the range whenever the user clicks on any line without holding `shift`. Then, when the user clicks another line while holding `shift`, it updates the start of the range if it was before the previously clicked line number or the end of the range if it was after. It does not perform the "move the edge closest to the selection" procedure. I did not intend to change this behavior though, since I considered it as different implementation details chosen by the implementer. It is not really necessary to copy the behavior of the GitHub UI anyway.

**Comment by tleb** [comment]
> GitHub UI sets both the start and the end of the range whenever the user clicks on any line without holding `shift`. Then, when the user clicks another line while holding `shift`, it updates the start of the range if it was before the previously clicked line number or the end of the range if it was after. It does not perform the "move the edge closest to the selection" procedure. I did not intend to change this behavior though, since I considered it as different implementation details chosen by the respective implementers. It is not really necessary to exactly copy the behavior of the GitHub UI anyway.

We discussed with @fstachura about the implementation we wanted. We picked one that felt more natural. Github's one has internal state that you need to remember and that is invisible to the user. Not that it matters much anyway.

---

I can confirm highlighting a single line is what we want; a poll inside Bootlin shows this is the preferred UX, by a large margin.

Can also confirm this will have to be rebased upon #338. I'll sync with @fstachura to get it merged swiftly. I had a review which needs discussing.

Thanks @jamestiotio!

**Comment by jamestiotio** [comment]
> We discussed with @fstachura about the implementation we wanted. We picked one that felt more natural. Github's one has internal state that you need to remember and that is invisible to the user. Not that it matters much anyway.

Actually, the internal states being tracked by the GitHub UI are exactly the same as the ones that Elixir currently tracks, it's just the logic that is slightly different. I _accidentally_ implemented the behavior of GitHub UI when I was experimenting with Elixir's code to make the changes for this PR. That's how I discovered the way they implemented their code highlighting logic. 😄

No worries though, I don't believe there is a right or wrong answer here. It is up to the Bootlin team to select whichever implementation suits Elixir the best and I will respect that decision.

> I can confirm highlighting a single line is what we want; a poll inside Bootlin shows this is the preferred UX, by a large margin.
> 
> Can also confirm this will have to be rebased upon #338. I'll sync with @fstachura to get it merged swiftly. I had a review which needs discussing.

Sounds great! Will make the necessary rebase and changes then.

**Comment by tleb** [comment]
#338 has been merged, do you have some time in the upcoming future to rebase? Thanks!

**Comment by jamestiotio** [comment]
@tleb I have rebased on the latest commit of the master branch and incremented the version number of `script.js`. I have also updated the commit message accordingly. It looks like the script is much better now with less code duplication.

Feel free to test it out and let me know your comments!

**Comment by tleb** [comment]
Tested, merged as f2146891c9cdfcfb2980cad44552ac309f1a975f & deployed, thanks @jamestiotio!
